src/storage/tile_group.cpp

```diff
@@ -314,6 +317,27 @@ oid_t TileGroup::InsertTupleFromCheckpoint(oid_t tuple_slot_id,
   return tuple_slot_id;
 }
 
+void TileGroup::CompressTiles() {
+  oid_t num_tiles = tiles.size();
+  for (oid_t i = 0; i < num_tiles; i++) {
+    LOG_TRACE("Compressing Tile %d in Tile Group", i);
+    Tile *old_tile = &*tiles[i];
```

Comment:
The correct way to get the pointer from `shared_ptr` is to use `shared_ptr::get` function.
This can be rewritten as `Tile *old_tile = tiles[i].get()`.